microsoft/testsuites/network/netinterface.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import time
+
+from assertpy import assert_that
+
+from lisa import (
+    LisaException,
+    Logger,
+    Node,
+    SkippedException,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    schema,
+    simple_requirement,
+)
+from lisa.features import NetworkInterface
+from lisa.nic import Nics
+from lisa.operating_system import CoreOs
+from lisa.tools import Dhclient, Uname, Wget
+from lisa.util import perf_timer
+
+
+@TestSuiteMetadata(
+    area="network",
+    category="functional",
+    description="""
+    This test suite validates basic functionalities of Network Interfaces.
+    """,
+)
+class NetInterface(TestSuite):
+    NETVSC_RELOAD_TEST_COUNT = 10
+    NET_INTERFACE_RELOAD_TEST_COUNT = 4
+    DHCLIENT_TIMEOUT = 15
+
+    @TestCaseMetadata(
+        description="""
+            This test case verifies if synthetic network module - netvsc
+                can be reloaded gracefully when done multiple times.
+
+            Steps:
+            1. Validate netvsc isn't built-in already. If it is then skip the test.
+            2. Unload and load netvsc module multiple times in a loop.
+
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            min_nic_count=1,
```

Comment:
This condition is not necessary. It's impossible to have no nic.

Reply:
We already have it in dpdk testsuite and synthetic testsuite and dnt see any harm in specifying it explicitly since we have test case where we boot up VM without adding any nic